specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2022-11-15-preview/managedCassandra.json

```diff
@@ -774,657 +693,1717 @@
+            },
+            "repairEnabled": {
+              "type": "boolean",
+              "description": "Should automatic repairs run on this cluster? If omitted, this is true, and should stay true unless you are running a hybrid cluster where you are already doing your own repairs."
+            },
+            "clientCertificates": {
+              "description": "List of TLS certificates used to authorize clients connecting to the cluster. All connections are TLS encrypted whether clientCertificates is set or not, but if clientCertificates is set, the managed Cassandra cluster will reject all connections not bearing a TLS client certificate that can be validated from one or more of the public certificates in this property.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Certificate"
+              }
+            },
+            "externalGossipCertificates": {
+              "description": "List of TLS certificates used to authorize gossip from unmanaged data centers. The TLS certificates of all nodes in unmanaged data centers must be verifiable using one of the certificates provided in this property.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Certificate"
+              }
+            },
+            "gossipCertificates": {
+              "readOnly": true,
+              "x-ms-mutability": [
+                "read"
+              ],
+              "description": "List of TLS certificates that unmanaged nodes must trust for gossip with managed nodes. All managed nodes will present TLS client certificates that are verifiable using one of the certificates provided in this property.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Certificate"
+              }
+            },
+            "externalSeedNodes": {
+              "type": "array",
+              "description": "List of IP addresses of seed nodes in unmanaged data centers. These will be added to the seed node lists of all managed nodes.",
+              "items": {
+                "$ref": "#/definitions/SeedNode"
+              }
+            },
+            "seedNodes": {
+              "readOnly": true,
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/SeedNode"
+              },
+              "description": "List of IP addresses of seed nodes in the managed data centers. These should be added to the seed node lists of all unmanaged nodes."
+            },
+            "hoursBetweenBackups": {
+              "type": "integer",
+              "format": "int32",
+              "description": "(Deprecated) Number of hours to wait between taking a backup of the cluster."
+            },
+            "deallocated": {
+              "type": "boolean",
+              "description": "Whether the cluster and associated data centers has been deallocated."
+            },
```

Comment:
ARM recommends enums over booleans for future proof APIs.

Standard guidance is: replace boolean/switch properties with better enum whenever possible.

A boolean will forever have two valid values (true or false). A string enum is more flexible and future proof option because it allows additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].. Also, properties should always ideally provide more meaningful values just than True and False, which are meaningless without context. For example two boolean switches "isTypeA" and "isTypeB" can be replaced with one enum "type": [A, B, DefaultType].

Note: do NOT just define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its still confusing, thanks to the boolean value names. Try to choose values that are meaningful even without context, like 'Encrypted', 'Unencrypted', 'TlsEncrypted'"